src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs

```diff
@@ -1660,5 +1660,25 @@ public static void Run()
     }
 }");
         }
+
+        [WorkItem(40644, "https://github.com/dotnet/roslyn/issues/40644")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task ShouldNotWarnForDataMemberFieldsInDataContractClasses()
```

Comment:
💡 Should include a test for a case where a class contains a field marked `[DataMember]` and a field which is not. The field without the attribute should still get marked `readonly`.

Reply:
➡️ Now fixed